scripts/exp/release.js

```diff
@@ -0,0 +1,259 @@
+
+async function publishExpPackages() {
+  try {
+    /**
+     * Welcome to the firebase release CLI!
+     */
+    console.log('Welcome to the Firebase Exp Packages release CLI!');
+
+    /**
+     * build packages
+     */
+    await buildPackages();
+
+    /**
+     * run tests
+     */
+    await runTests();
+
+    // path to exp packages
+    const packagePaths = await mapWorkspaceToPackages([
+      `${projectRoot}/packages-exp/*`
+    ]);
+
+    /**
+     * It does 2 things:
+     *
+     * 1. Bumps the pactch version of exp packages regardless if there is any update
+     * since the last release. This simplifies the script and works fine for exp packages.
+     *
+     * 2. Removes -exp in package names because we will publish them using
+     * the existing package names under a special release tag (e.g. firebase@exp).
+     */
+    const versions = await updatePackageNamesAndVersions(packagePaths);
+
+    /**
+     * Release packages to NPM
+     */
+    await publishToNpm(packagePaths);
+
+    /**
+     * reset the working tree to recover package names with -exp in the package.json files,
+     * then bump patch version of all exp packages
+     */
+    await resetWorkingTreeAndBumpVersions(packagePaths, versions);
+
+    /**
+     * push to github
+     */
+    await commitAndPush(versions);
+  } catch (err) {
+    /**
+     * Log any errors that happened during the process
+     */
+    console.error(err);
+
+    /**
+     * Exit with an error code
+     */
+    process.exit(1);
+  }
+}
+
+async function buildPackages() {
+  const spinner = ora(' Building Packages').start();
+  await spawn('yarn', ['build:exp'], {
+    cwd: projectRoot
+  });
+  spinner.stopAndPersist({
+    symbol: '✅'
+  });
+}
+
+async function runTests() {
+  await spawn('yarn', ['test:exp'], {
+    cwd: projectRoot,
+    stdio: 'inherit'
+  });
+}
+
+async function updatePackageNamesAndVersions(packagePaths) {
+  // get package name -> next version mapping
+  const versions = new Map();
+  for (const path of packagePaths) {
+    const { version, name } = await readPackageJson(path);
+    // increase the patch version of all exp packages
+    const nextVersion = inc(version, 'patch');
+    versions.set(name, nextVersion);
+  }
+
+  await updatePackageJsons(packagePaths, versions, {
+    removeExpInName: true,
+    updateVersions: true,
+    makePublic: true
+  });
+
+  return versions;
+}
+
+async function publishToNpm(packagePaths) {
+  const taskArray = await Promise.all(
+    packagePaths.map(async pp => {
+      const { version, name } = await readPackageJson(pp);
+      return {
+        title: `📦  ${name}@${version}`,
+        task: () => publishPackage(pp)
+      };
+    })
+  );
+
+  const tasks = new Listr(taskArray, {
+    concurrent: false,
+    exitOnError: false
+  });
+
+  console.log('\r\nPublishing Packages to NPM:');
+  return tasks.run();
+}
+
+async function publishPackage(packagePath) {
+  // TODO: remove --dry-run
```

Comment:
Reminder to revert.

Reply:
Will remove once launchcal is approved. For now, I will keep `--dry-run` to prevent from publishing to npm accidentally.